backend/src/main/java/com/tradomator/service/WalletService.java

```diff
@@ -0,0 +1,40 @@
+package com.tradomator.service;
+
+import com.tradomator.model.Wallet;
+import com.tradomator.repo.WalletRepo;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class WalletService {
+
+    private final WalletRepo walletRepo;
+
+
+    public WalletService(WalletRepo walletRepo) {
+        this.walletRepo = walletRepo;
+
+    }
+
+
+    public Wallet getLatestWalletDocument() {
+        List<Wallet> all = walletRepo.findAll();
```

Comment:
The logic can be moved to the database. It should be possible to define a method in the repository that looks like that:
`Wallet findTopByOrderByUpdateTimeDesc()`
The MongoRepository should generate the query based on the method name. That way only one element is retrieved from the database and instantiated and not all of them (which will become expensive as the database grows).

Reply:
that's true. Brilliant. Already had this thought that it would become a performance issue. So future proof solution. Thanks.

Reply:
also nice one, defining the name the way you did enables suggestions from Intellij :-)

Reply:
wow, works and is really fast. Thanks a lot appreciated!